x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -795,10 +797,18 @@ const updateExistingDataStream = async ({
 
     // if update fails, rollover data stream and bail out
   } catch (err) {
-    logger.info(`Mappings update for ${dataStreamName} failed due to ${err}`);
-    logger.info(`Triggering a rollover for ${dataStreamName}`);
-    await rolloverDataStream(dataStreamName, esClient);
-    return;
+    if (
+      isResponseError(err) &&
+      err.statusCode === 400 &&
+      err.body?.error?.type === 'illegal_argument_exception'
+    ) {
+      logger.info(`Mappings update for ${dataStreamName} failed due to ${err}`);
+      logger.info(`Triggering a rollover for ${dataStreamName}`);
+      await rolloverDataStream(dataStreamName, esClient);
+      return;
+    }
+    logger.error(`Mappings update for ${dataStreamName} failed due to unexpected error: ${err}`);
+    throw err;
```

Comment:
My only concern here is that this might surface existing errors that were ignored before, but I think it's better to see the errors early.